File: test/src/main/java/com/jilaba/test/service/ProductService.java
package com.jilaba.test.service;

import com.jilaba.test.model.CommonModel;
import com.jilaba.test.model.Product;

import java.util.List;

public interface ProductService {

    public List<CommonModel> getProductGroup() throws Exception;

    public List<CommonModel> getQuality() throws Exception;

    public List<CommonModel> getStyle() throws Exception;

    public List<CommonModel> getSize() throws Exception;

    public List<CommonModel> getSizeGroup() throws Exception;

    public List<CommonModel> getProduct() throws Exception;

    public List<CommonModel> getBrand() throws Exception;

    public List<CommonModel> getHsn() throws Exception;

    public void saveProduct(Product product) throws Exception;

    public List<Product> getProductView() throws Exception;

    public void deleteProduct(String productCode) throws Exception;

    public void updateProduct(Product product) throws Exception;
}

File: textapi/src/app/service/product.service.ts
import { HttpClient } from '@angular/common/http';
import { inject, Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { product } from '../model/product';
import { api } from '../appconstant';

@Injectable({
  providedIn: 'root'
})
export class ProductService {

  httpClient: HttpClient = inject(HttpClient)
  productEntryApi = api + '/product/entry/'

  getProductGroup(): Observable<product[]> {
    return this.httpClient.get<product[]>(this.productEntryApi + "productgroup");
  }

  getProduct(): Observable<product[]> {
    return this.httpClient.get<product[]>(this.productEntryApi + "product");
  }

  getSize(): Observable<product[]> {
    return this.httpClient.get<product[]>(this.productEntryApi + "size");
  }

  getSizeGroup(): Observable<product[]> {
    return this.httpClient.get<product[]>(this.productEntryApi + "sizegroup");
  }

  getQuality(): Observable<product[]> {
    return this.httpClient.get<product[]>(this.productEntryApi + "quality");
  }

  getStyle(): Observable<product[]> {
    return this.httpClient.get<product[]>(this.productEntryApi + "style");
  }

  getBrand(): Observable<product[]> {
    return this.httpClient.get<product[]>(this.productEntryApi + 'brand');
  }

  getHsn(): Observable<product[]> {
    return this.httpClient.get<product[]>(this.productEntryApi + 'hsn');
  }

   async saveProduct( product:any):Promise<Observable<any>>{
    const urls = this.productEntryApi + 'save'
    return await this.httpClient.post(urls,product);
  }

  getProductView():Observable<product[]>{
    return this.httpClient.get<product[]>(this.productEntryApi+'viewproduct')
  }

}
